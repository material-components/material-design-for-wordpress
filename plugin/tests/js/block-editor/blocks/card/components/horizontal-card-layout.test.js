/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import HorizontalCardLayout from '../../../../../../assets/src/block-editor/blocks/card/components/horizontal-card-layout';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <HorizontalCardLayout { ...props } /> );
};

const baseProps = {
	cardIndex: 0,
	contentLayout: 'text-under-media',
	title: 'Test title',
	displayTitle: true,
	secondaryText: 'Test secondary text',
	displaySecondaryText: true,
	imageSourceUrl: 'http://test.loc/test.jpg',
	isImageEditMode: false,
	displayImage: true,
	primaryActionButtonLabel: 'Test button action 1',
	primaryActionButtonUrl: 'http://test.loc/hello-world',
	primaryActionButtonNewTab: true,
	primaryActionButtonNoFollow: true,
	secondaryActionButtonLabel: 'Test button action 2',
	secondaryActionButtonUrl: 'http://test.loc/hello-world-again',
	secondaryActionButtonNewTab: true,
	secondaryActionButtonNoFollow: true,
	displaySecondaryActionButton: true,
	displayActions: true,
	outlined: true,
	cornerRadius: 4,
	setter: jest.fn(),
	isEditMode: false,
};

describe( 'HorizontalCardLayout', () => {
	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the image should not be displayed', () => {
		const props = { ...baseProps };
		props.displayImage = false;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the action row should not be displayed', () => {
		const props = { ...baseProps };
		props.displayActions = false;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the corner radius is undefined', () => {
		const props = { ...baseProps };
		props.cornerRadius = undefined;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
