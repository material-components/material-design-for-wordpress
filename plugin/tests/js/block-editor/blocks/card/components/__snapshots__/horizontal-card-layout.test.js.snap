// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalCardLayout matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action material-design-card__primary-action"
          tabindex="0"
        >
          <div
            class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
            style="background-image: url(http://test.loc/test.jpg);"
            tabindex="0"
          />
          <div
            class="material-design-card__primary"
          >
            <h2
              class="material-design-card__title mdc-typography mdc-typography--headline6"
            >
              Test title
            </h2>
            <h3
              class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
            >
              Test secondary text
            </h3>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 1
              </span>
            </a>
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world-again"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 2
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
      style="border-radius: 4px;"
    >
      <div
        class="mdc-card__primary-action material-design-card__primary-action"
        tabindex="0"
      >
        <div
          class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
          style="background-image: url(http://test.loc/test.jpg);"
          tabindex="0"
        />
        <div
          class="material-design-card__primary"
        >
          <h2
            class="material-design-card__title mdc-typography mdc-typography--headline6"
          >
            Test title
          </h2>
          <h3
            class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
          >
            Test secondary text
          </h3>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 1
            </span>
          </a>
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world-again"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 2
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HorizontalCardLayout matches snapshot when the action row should not be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action material-design-card__primary-action"
          tabindex="0"
        >
          <div
            class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
            style="background-image: url(http://test.loc/test.jpg);"
            tabindex="0"
          />
          <div
            class="material-design-card__primary"
          >
            <h2
              class="material-design-card__title mdc-typography mdc-typography--headline6"
            >
              Test title
            </h2>
            <h3
              class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
            >
              Test secondary text
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
      style="border-radius: 4px;"
    >
      <div
        class="mdc-card__primary-action material-design-card__primary-action"
        tabindex="0"
      >
        <div
          class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
          style="background-image: url(http://test.loc/test.jpg);"
          tabindex="0"
        />
        <div
          class="material-design-card__primary"
        >
          <h2
            class="material-design-card__title mdc-typography mdc-typography--headline6"
          >
            Test title
          </h2>
          <h3
            class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
          >
            Test secondary text
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HorizontalCardLayout matches snapshot when the corner radius is undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
      >
        <div
          class="mdc-card__primary-action material-design-card__primary-action"
          tabindex="0"
        >
          <div
            class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
            style="background-image: url(http://test.loc/test.jpg);"
            tabindex="0"
          />
          <div
            class="material-design-card__primary"
          >
            <h2
              class="material-design-card__title mdc-typography mdc-typography--headline6"
            >
              Test title
            </h2>
            <h3
              class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
            >
              Test secondary text
            </h3>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 1
              </span>
            </a>
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world-again"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 2
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
    >
      <div
        class="mdc-card__primary-action material-design-card__primary-action"
        tabindex="0"
      >
        <div
          class="mdc-card__media mdc-card__media--square material-design-card__media material-design-card-with-text-under-media"
          style="background-image: url(http://test.loc/test.jpg);"
          tabindex="0"
        />
        <div
          class="material-design-card__primary"
        >
          <h2
            class="material-design-card__title mdc-typography mdc-typography--headline6"
          >
            Test title
          </h2>
          <h3
            class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
          >
            Test secondary text
          </h3>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 1
            </span>
          </a>
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world-again"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 2
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`HorizontalCardLayout matches snapshot when the image should not be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <p
      class="a11y-speak-intro-text"
      hidden="hidden"
      id="a11y-speak-intro-text"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    >
      Notifications
    </p>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
        style="border-radius: 4px;"
      >
        <div
          class="mdc-card__primary-action material-design-card__primary-action"
          tabindex="0"
        >
          <div
            class="material-design-card__primary"
          >
            <h2
              class="material-design-card__title mdc-typography mdc-typography--headline6"
            >
              Test title
            </h2>
            <h3
              class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
            >
              Test secondary text
            </h3>
          </div>
        </div>
        <div
          class="mdc-card__actions"
        >
          <div
            class="mdc-card__action-buttons"
          >
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 1
              </span>
            </a>
            <a
              class="mdc-button mdc-card__action mdc-card__action--button"
              href="http://test.loc/hello-world-again"
              rel="noopener noreferrer nofollow"
              target="_blank"
            >
              <div
                class="mdc-button__ripple"
              />
              <span
                class="mdc-button__label"
              >
                Test button action 2
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card mdc-card--outlined material-design-card material-design-card__list material-design-basic"
      style="border-radius: 4px;"
    >
      <div
        class="mdc-card__primary-action material-design-card__primary-action"
        tabindex="0"
      >
        <div
          class="material-design-card__primary"
        >
          <h2
            class="material-design-card__title mdc-typography mdc-typography--headline6"
          >
            Test title
          </h2>
          <h3
            class="material-design-card__secondary-text mdc-typography mdc-typography--subtitle2"
          >
            Test secondary text
          </h3>
        </div>
      </div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 1
            </span>
          </a>
          <a
            class="mdc-button mdc-card__action mdc-card__action--button"
            href="http://test.loc/hello-world-again"
            rel="noopener noreferrer nofollow"
            target="_blank"
          >
            <div
              class="mdc-button__ripple"
            />
            <span
              class="mdc-button__label"
            >
              Test button action 2
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
